### PR TITLE
AN-66: add preprocessing script

### PR DESCRIPTION
**Pull request by tyagiprnv**

### CODE REVIEW

**Review by lkno0705 — COMMENTED** [review]
- What are these eval results for / from?
- Notebook is a bit unstructured, as there are no real explanations on what is done, so a short comment on hard to read code would be nice for the future but its ok
- All of the contents is for the old Gold Standard, which isn't used for the final results, so what about the rest?
- Does the Fine-Tuning script remain unchanged ?

**Comment by tyagiprnv** [comment]
1. The results are from T5 trained on the new labelled data which is why I put it in a new folder T5_master.
2. I will add comments where needed.
3. I pushed this a long time ago when we only had the gold standard, I will add the preprocessing script for new data.
4. Yes the fine-tuning script for T5 remains unchanged since I formatted master_label data just like the goldstandard i.e. the same format fine-tuning script required.

**Review by Niklas-23 — CHANGES_REQUESTED** [review]
Please add the new preprocessing script and use meaningful names for your variables/dataframes.

**Comment by tyagiprnv** [comment]
Just did, Also restructured the folder.

File: notebooks/models/text-to-text/T5/master_processing.ipynb [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+     "start_time": "2023-08-03T12:29:53.318773Z"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "outputs": [],
+   "source": [
+    "melted_df.rename(columns={'Marked for processing':'MFP','Value':'question'},inplace=True)"
```
Comment (Niklas-23) [inline notebooks/models/text-to-text/T5/master_processing.ipynb]: You can rename this columns when you use the melt function

File: notebooks/models/text-to-text/T5/master_processing.ipynb [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"[^A-Za-z0-9(),!?@\\'\\`\\\"\\_\\n]\", \" \")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"@\", \"at\")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace('\\n','', regex=True)\n",
+    "    dff1[text_field] = dff1[text_field].str.lower()\n",
+    "    return dff1[text_field]\n",
+    "\n",
+    "merged_df['context'] = standardize_text(merged_df, \"context\")\n",
+    "merged_df['Question 1'] = merged_df['Question 1'].str.lower()\n",
+    "merged_df['Question 2'] = merged_df['Question 2'].str.lower()\n",
+    "merged_df['Question 3'] = merged_df['Question 3'].str.lower()"
```
Comment (Niklas-23) [inline notebooks/models/text-to-text/T5/master_processing.ipynb]: Isn't T5 case sensitive?

File: notebooks/models/text-to-text/T5/master_processing.ipynb [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "outputs": [],
+   "source": [
+    "# reading extracted content saved in seperate files.\n",
+    "data_all = []\n",
+    "for file in lecture_list:\n",
+    "    df = pd.read_csv(f'data/{file}.csv')\n",
+    "    df['PDF-Name'] = f'{file}.pdf'\n",
```
Comment (Niklas-23) [inline notebooks/models/text-to-text/T5/master_processing.ipynb]: Please add the code you used to create these .csv file

File: notebooks/models/text-to-text/T5/master_processing.ipynb [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+     "start_time": "2023-08-03T12:29:30.477792Z"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "outputs": [],
+   "source": [
+    "content_df.rename(columns={'PDF-Name':'Name'},inplace=True)"
```
Comment (Niklas-23) [inline notebooks/models/text-to-text/T5/master_processing.ipynb]: You don't need to rename the column here. You can do it when you read the .csv file

**Review by Niklas-23 — CHANGES_REQUESTED** [review]
In general, please combine code cells where suitable

File: notebooks/models/text-to-text/T5/master_processing.ipynb (hunk 1 above)
Comment (tyagiprnv) [reply notebooks/models/text-to-text/T5/master_processing.ipynb]: done

File: notebooks/models/text-to-text/T5/master_processing.ipynb (hunk 2 above)
Comment (tyagiprnv) [reply notebooks/models/text-to-text/T5/master_processing.ipynb]: Yes, it is. Do you reckon I should not lowercase all text? I thought it wouldn't make a difference and it is usually a standard practice in nlp. I can of course not do it if it seems beneficial to you.

File: notebooks/models/text-to-text/T5/master_processing.ipynb (hunk 3 above)
Comment (tyagiprnv) [reply notebooks/models/text-to-text/T5/master_processing.ipynb]: done

File: notebooks/models/text-to-text/T5/master_processing.ipynb (hunk 4 above)
Comment (tyagiprnv) [reply notebooks/models/text-to-text/T5/master_processing.ipynb]: done

File: notebooks/models/text-to-text/T5/master_processing.ipynb (hunk 2 above)
Comment (Niklas-23) [reply notebooks/models/text-to-text/T5/master_processing.ipynb]: It was practice because in the past the models have not distinguished but for new and case sensitive models, of course, it is important. I think you have to run the fine-tuning again with the corrected training data.

File: notebooks/models/text-to-text/T5/master_processing.ipynb [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "#cleaning text\n",
+    "def standardize_text(dff1, text_field):\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"http\\S+\", \"\")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"http\", \"\")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"@\\S+\", \"\")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"[^A-Za-z0-9(),!?@\\'\\`\\\"\\_\\n]\", \" \")\n",
```
Comment (Niklas-23) [inline notebooks/models/text-to-text/T5/master_processing.ipynb]: Why do you use this regex? Honestly, I personally would leave punctuation removal to the tokenizer. For example tokens like [] or {} are ignored/removed here

File: notebooks/models/text-to-text/T5/master_processing.ipynb (hunk 2 above)
Comment (tyagiprnv) [reply notebooks/models/text-to-text/T5/master_processing.ipynb]: alright

**Review by Niklas-23 — CHANGES_REQUESTED** [review]

File: notebooks/models/text-to-text/T5/master_processing.ipynb (hunk 5 above)
Comment (tyagiprnv) [reply notebooks/models/text-to-text/T5/master_processing.ipynb]: I looked it up, it is still common practice to remove this stuff, but for our dataset we can maybe argue in favor of keeping them since they might provide extra context given the nature of slides. So I'll remove this regex.

Comment (tyagiprnv) [reply notebooks/models/text-to-text/T5/master_processing.ipynb]: Yes, So I removed the regex and fine-tuned again. Also let's keep the results from earlier as well just for comparison. And removing most regex didn't improve metric but surely improved quality of some questions so yes this definitely Is the right way to approach preprocessing

**Review by Niklas-23 — APPROVED** [review]
Now it looks good!
